wp-content/themes/goonj-crm/functions.php

```diff
@@ -270,17 +279,29 @@ function goonj_handle_user_identification_form() {
 			exit;
 		}
 
-		// If we are here, then it means for sure that the contact exists.
+		$allowedPurposes = ['material-contribution', 'dropping-center-contribution'];
 
-		if ( 'material-contribution' === $purpose ) {
-			$material_contribution_form_path = sprintf(
-				'/material-contribution/#?email=%s&phone=%s&Material_Contribution.Collection_Camp=%s&source_contact_id=%s',
-				$email,
-				$phone,
-				$target_id,
-				$found_contacts['id']
-			);
-			wp_redirect( $material_contribution_form_path );
+		// If we are here, then it means for sure that the contact exists.
+		if (in_array($purpose, $allowedPurposes)) {
```

Comment:
What is the need for this check when inside this we have more strict checks:
1. `$purpose === 'material-contribution'`
2. `$purpose === 'dropping-center-contribution'`

Is there any case that is breaking if we don't use `in_array($purpose, $allowedPurposes)`?

Reply:
Yes , we don't need this now, I have updated the code. 
```
if (in_array($purpose, $allowedPurposes)) {
```